### ISSUE TITLE
Missing UEFI support (Bugzilla Bug 568)

### ISSUE THREAD

**Issue by tpgxyz** [OpenMandrivaAssociation/test2#568]
This issue was created automatically with bugzilla2github
# Bugzilla Bug 568
Date: 2014-03-13 14:58:10 +0000
From: @tpgxyz
To:   @berolinux
CC:   efremmc2@gmail.com, gio_6b@yahoo.it, ishneww@gmail.com, @itchka, jeremielapuree@yahoo.fr, @robxu9

Depends on:   #390
See also:     https://issues.openmandriva.org/show_bug.cgi?id=390
See also:     https://issues.openmandriva.org/show_bug.cgi?id=465
See also:     https://issues.openmandriva.org/show_bug.cgi?id=492
See also:     https://issues.openmandriva.org/show_bug.cgi?id=355
Last updated: 2014-05-06 07:07:30 +0000

## Comment 3876
Date: 2014-03-13 14:58:10 +0000
From: @tpgxyz

As you may have noticed OMV Lx is not supporting UEFI. This is a major feature lack.

Hopefully this will be fixed for next release than 2014.0

## Comment 3877
Date: 2014-03-13 15:00:31 +0000
From: @tpgxyz

This bug is created to cover all UEFI support related bugs.
Current list of bugs:
390
465
492

## Comment 3879
Date: 2014-03-13 15:01:58 +0000
From: @tpgxyz

*** Bug #492 has been marked as a duplicate of this bug. ***

## Comment 3881
Date: 2014-03-13 15:02:33 +0000
From: @tpgxyz

*** Bug #390 has been marked as a duplicate of this bug. ***

## Comment 3883
Date: 2014-03-13 15:02:45 +0000
From: @tpgxyz

*** Bug #465 has been marked as a duplicate of this bug. ***

## Comment 3884
Date: 2014-03-13 15:04:28 +0000
From: @tpgxyz

Assigning this bug to Bero.

## Comment 3964
Date: 2014-03-14 13:59:09 +0000
From: @tpgxyz

*** Bug #355 has been marked as a duplicate of this bug. ***

## Comment 4012
Date: 2014-03-15 18:05:53 +0000
From: jeremielapuree@yahoo.fr

What do you mean by "Hopefully this will be fixed for next release than 2014.0"?

UEFI support is not foreseen for Openmandriva 2014.0?

## Comment 4032
Date: 2014-03-16 14:59:38 +0000
From: @tpgxyz

(In reply to comment #7)
> What do you mean by "Hopefully this will be fixed for next release than
> 2014.0"?
> 
> UEFI support is not foreseen for Openmandriva 2014.0?

Due to complexity of this task we decided to postpone this feature to next release after 2014.0

## Comment 4229
Date: 2014-03-21 15:53:31 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

Bad news. This means that at least for several months to come any Windows 8 user that want try or use OpenMandriva cannot have a dual boot system.
Is it so complicated to follow the same way already used by other distributions? 

In my opinion this is one of the first bug to solve for a new distribution.

## Comment 4287
Date: 2014-03-23 18:32:01 +0000
From: @robxu9

(In reply to comment #9)
> Bad news. This means that at least for several months to come any Windows 8
> user that want try or use OpenMandriva cannot have a dual boot system.
> Is it so complicated to follow the same way already used by other
> distributions? 
> 
> In my opinion this is one of the first bug to solve for a new distribution.

Yet it is not so easy. We need to adapt our tools to use UEFI standards. We need to determine whether we should pay to get our keys signed by Secure Boot.

Those are big decisions that affect many core packages, and so we're postponing it.

## Comment 4291
Date: 2014-03-23 21:04:53 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

(In reply to comment #10)
> Yet it is not so easy. We need to adapt our tools to use UEFI standards. 
This need a big amount of work I suppose.

> We need to determine whether we should pay to get our keys signed by Secure
> Boot.
I'm not an expert but I can assure that a friend is using a very known Linux distribution with "Secure boot" disabled. And his dual boot system with Windows 8 and Linux GRUB works normally. May be this can avoid paying for Secure Boot.

## Comment 5141
Date: 2014-04-15 21:21:00 +0000
From: @tpgxyz

Starting 2014.0 RC1 we will have EFI support working, please do testing.

## Comment 5142
Date: 2014-04-15 21:40:00 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

I'm already testing.
I tried to install from usb. Situation is improved a bit but after few seconds installation stops showing this message:

OpenMandriva Lx release 2014.0 (Phosporus) for x86_64
kernel 3.13.9-nrjQL-desktop-1omv on a 8-processor x86_64/tty1
localhost login:

From here I can login as root and open Midnight Commander. If I give "startx" graphic interface don't opens.
If you need more information please tell me how to search for.

## Comment 5351
Date: 2014-04-23 06:28:50 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

tried RC1, installation stops with

dracut:/#

## Comment 5352
Date: 2014-04-23 06:36:14 +0000
From: @tpgxyz

(In reply to comment #13) 
> From here I can login as root and open Midnight Commander. If I give
> "startx" graphic interface don't opens.

This looks like nothing related to EFI. What graphics card you you have? Ati ?

## Comment 5360
Date: 2014-04-23 07:16:12 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

(In reply to comment #15)
> (In reply to comment #13) 
> > From here I can login as root and open Midnight Commander. If I give
> > "startx" graphic interface don't opens.
This happened with previous release.
Now I'm trying RC1 released on 18-4 and it stops with "dracut" prompt. See comment 14

> 
> This looks like nothing related to EFI. What graphics card you you have? 
From windows control panel it's AMD Radeon HD7600M

If you need more information please tell me how to search for.

## Comment 5372
Date: 2014-04-23 11:26:05 +0000
From: @tpgxyz

(In reply to comment #16)

> 
> If you need more information please tell me how to search for.

Please attach output from omv-bug-report.sh (do not run startx)

## Comment 5380
Date: 2014-04-23 14:49:22 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

(In reply to comment #17)
> Please attach output from omv-bug-report.sh (do not run startx)
From "dracut" prompt I don't know how to do.
I tried Live too but unfortunately this also stop with "dracut" prompt.

## Comment 5399
Date: 2014-04-23 23:45:25 +0000
From: @itchka

Giorgio,
1. At the dracut prompt we need find out what is going on.
First try typing "exit" and if there is an error message please tell us what it is.

2. Next I need to find out what is working and what is not. So after doing step 1 reboot and at the dracut prompt please plug in a memory stick. Insert it into another USB slot if you have just booted the live version from an already plugged in memory stick..
 
3. Type "dmesg" (without the quotes) and observe the output. At the very end you should see a reference like [sdb] or [sdd] make note of it.

4.  Use:  mkdir /mnt
5.  Use:  mount -t vfat /dev/sd<x> the name from 3 above 
6. A command to make things a little easier. Type the following.
ln -s /sbin/modprobe /sbin/lsmod

7. Now we will save some interesting data. 

Use the following commands to create a files on your usb stick:
journalctl --no-pager > /mnt/jnl.txt
lsmod > /mnt/mods

If an rdsreport is mentioned during boot please copy that also to the /mnt directory.

When all is complete type "Unmount /mnt" at which point you can unplug the usb stick.

Please attach the files collected to this bug and I will try and determine what the problem is.

## Comment 5676
Date: 2014-04-29 10:43:55 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

(In reply to comment #19)
> Giorgio,
> 1. At the dracut prompt we need find out what is going on.
> First try typing "exit" and if there is an error message please tell us what
> it is.

This is what I get. I copied the text because I can't save it.

dracut:/# exit
[  OK  ] Started Dracut Emergency Shell.
[ 128.000247] dracut-mount[482]: mount: unknown filesystem type 'DM_snapshot_cow'
[ 128.001113] dracut-mount[482]: Warning: Can't mount root filesystem
		Starting Dracut Emergency Shell...

Generating “/run/initramfs/rdosreport.txt”

Entering emergency mode. Exit the shell to continue.
Type “journalctl” to view system logs.
You might want to save “/run/initramfs/rdosreport.txt” to a USB stick or /boot 
after mounting them and attach it to a bug report.

## Comment 5680
Date: 2014-04-29 11:44:43 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

(In reply to comment #19)

> 2. Next I need to find out what is working and what is not. So after doing
> step 1 reboot and at the dracut prompt please plug in a memory stick. Insert
> it into another USB slot if you have just booted the live version from an
> already plugged in memory stick..
Done.

> 3. Type "dmesg" (without the quotes) and observe the output. At the very end
> you should see a reference like [sdb] or [sdd] make note of it.
My usb stick is sdc

 
> 4.  Use:  mkdir /mnt
mnt folder is created, I checked with ls command. But where is this folder? In RAM?

> 5.  Use:  mount -t vfat /dev/sdc
mount: can't find /dev/sdc in /etc/fstab Something goes wrong. What? 

 
> 7. Use the following commands to create a files on your usb stick:
> journalctl --no-pager > /mnt/jnl.txt
> lsmod > /mnt/mods
Checked and found jnl.txt and mods in /mnt

> If an rdsreport is mentioned during boot please copy that also to the /mnt
> directory.
I had this file /run/initramfs/rdosreport.txt, from inside initramfs folder I give this command that don't works
cp  rdosreport.txt /mnt
cp: cannot stat ' rdosreport.txt': No such file or directory 

> When all is complete type "Unmount /mnt" at which point you can unplug the
> usb stick.
The right command "umount" I think.
umount /mnt and /mnt disappear.

 
> Please attach the files collected to this bug and I will try and determine
> what the problem is.
I don't find any file in usb stick.

Perhaps the error is in step 5 ?
In step 7 files are created where? In usb stick?

## Comment 5682
Date: 2014-04-29 12:31:45 +0000
From: @itchka

Closing

This bug has become corrupted with other issues. UEFI support is now enabled in OMV but is still considered ALPHA please report bugs as you find them.
The other issues will be rasied with the individual user by QA.

Colin Close
QA Team

## Comment 5683
Date: 2014-04-29 12:32:06 +0000
From: @itchka

Closing

This bug has become corrupted with other issues. UEFI support is now enabled in OMV but is still considered ALPHA please report bugs as you find them.
The other issues will be rasied with the individual user by QA.

Colin Close
QA Team

## Comment 5702
Date: 2014-04-29 14:35:43 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

So what?
May be I have to open a new bug like "installation stops with dracut prompt?"

## Comment 5705
Date: 2014-04-29 15:15:53 +0000
From: @tpgxyz

(In reply to comment #24)
> So what?
> May be I have to open a new bug like "installation stops with dracut prompt?"

Please be adives this bug is realted to UEFI support, not the boot process after EFI screen.

From you descriptions, this is cleary visible that EFI mechanisms works, but you fall into two different issues not related to EFI itself.

First one is with you graphics card https://issues.openmandriva.org/show_bug.cgi?id=568#c13

Second one with https://issues.openmandriva.org/show_bug.cgi?id=568#c20 which sound like you have broken usb stick or your ISO sha1sum is wrong or installation on usb stick went wrong.

To keep things sane, let's just stick to the topic.

## Comment 5876
Date: 2014-05-06 07:06:26 +0000
From: jeremielapuree@yahoo.fr

I did my homework. I tested Openmandriva 2014 with UEFI. And GREAT NEWS. Install process works very well.

So, I think that this bug can be closed as FIXED.

Note I have Windows 7, so no secure boot is requested. But anyway, if there is a problem with secure boot, it would deserve to open its own bug report.

Godd job, Openmandriva ;)

## Comment 5877
Date: 2014-05-06 07:07:30 +0000
From: jeremielapuree@yahoo.fr

Fixed
